packages/vuetify/src/components/VOtpInput/VOtpInput.ts

```diff
@@ -0,0 +1,343 @@
+// Styles
+import '../VTextField/VTextField.sass'
+import './VOtpInput.sass'
+
+// Extensions
+import VInput from '../VInput'
+
+// Directives
+import ripple from '../../directives/ripple'
+
+// Utilities
+import { convertToUnit, keyCodes } from '../../util/helpers'
+import { breaking } from '../../util/console'
+
+// Types
+import mixins from '../../util/mixins'
+import { VNode } from 'vue'
+
+const baseMixins = mixins(
+  VInput,
+)
+interface options extends InstanceType<typeof baseMixins> {
+  $refs: {
+    input: HTMLInputElement
+  }
+}
+
+/* @vue/component */
+export default baseMixins.extend<options>().extend({
+  name: 'v-otp-input',
+
+  directives: {
+    ripple,
+  },
+
+  inheritAttrs: false,
+
+  props: {
+    length: {
+      type: Number,
+      default: 6,
+    },
+    type: {
+      type: String,
+      default: 'text',
+    },
+  },
+
+  data: () => ({
+    badInput: false,
+    initialValue: null,
+    isBooted: false,
+    otp: [] as string[],
+  }),
+
+  computed: {
+    classes (): object {
+      return {
+        ...VInput.options.computed.classes.call(this),
+        'v-input--hide-details': true,
+        'v-text-field': true,
+        'v-otp-input': true,
+        'v-text-field--enclosed': true,
+        'v-text-field--outlined': true,
+      }
+    },
+    internalValue: {
+      get (): any {
+        return this.lazyValue
+      },
+      set (val: any) {
+        this.lazyValue = val
+        this.$emit('input', this.lazyValue)
+      },
+    },
+    isDirty (): boolean {
+      return this.lazyValue?.toString().length > 0 || this.badInput
+    },
+  },
+
+  watch: {
+    isFocused: 'updateValue',
+    value (val) {
+      this.lazyValue = val
+    },
+  },
+
+  created () {
+    /* istanbul ignore next */
+    if (this.$attrs.hasOwnProperty('browser-autocomplete')) {
+      breaking('browser-autocomplete', 'autocomplete', this)
+    }
+
+    this.otp = this.internalValue?.split('') || []
+  },
+
+  mounted () {
+    requestAnimationFrame(() => (this.isBooted = true))
+  },
+
+  methods: {
+    /** @public */
+    focus (e: Event, otpIdx: number) {
+      this.onFocus(e, otpIdx || 0)
+    },
+    genInputSlot (otpIdx: number) {
+      return this.$createElement('div', this.setBackgroundColor(this.backgroundColor, {
+        staticClass: 'v-input__slot',
+        style: { height: convertToUnit(this.height) },
+        on: {
+          click: () => this.onClick(otpIdx),
+          mousedown: (e: Event) => this.onMouseDown(e, otpIdx),
+          mouseup: (e: Event) => this.onMouseUp(e, otpIdx),
+        },
+        ref: `input-slot-${otpIdx}`,
```

Comment:
With `refInFor: true` the ref will be an array so you don't have to do all the weird string stuff

Reply:
Lesson learned here - ty!
Applying it.